### PR TITLE
[CDAP-11010] Fixes saving name of the pipeline to save the pipeline as draft

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA - https://issues.cask.co/browse/CDAP-11010

- Changed "Ok" button while editing name & description to be "Save" as it saves the pipeline as draft.
- Currently we only save the pipeline as draft when clicking on "Save" button. We don't support autosave yet.

Build - https://builds.cask.co/browse/CDAP-UDUT873/latest

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js [hunk 1]
```diff
@@ -212,6 +242,7 @@ class HydratorPlusPlusTopPanelCtrl {
     this.metadataExpanded = false;
     event.preventDefault();
     event.stopPropagation();
+    this.onSaveDraft();
```
Comment (njbriggs) [inline cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js:245]: Other than name, what else can trigger a `saveMetadata` event?

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js:245]: Right now changing name and description (part of metadata section) triggers the save. Other than that the user can click on "Save" on the top panel to save the pipeline as draft.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**